### ISSUE TITLE
Rename DeepStruct to Config

### ISSUE THREAD

**Issue by martinsvalin** [klarna/phobos#13]
To make sure DeepStruct doesn't end up being used outside its current purpose without some thought given, I propose we rename the class to Config.

The implementation hasn't been thought through to cover performance or security issues if it's used on incoming data. Using it on config files should be totally fine.

**Comment by klippx** [comment]
I think it can stay as DeepStruct because that is what it is. It is based on OpenStruct, and is not doing much on top of it tbh.

OpenStruct itself uses `:to_sym` in 6 different places in ruby 2.3, and in 4 different places in 2.1.x (havent bothered to check earlier). This means ruby developers should be careful with using either OpenStruct or DeepStruct. They are dynamic and as such have their concerns. 

From documentation of OpenStruct

```
# An OpenStruct utilizes Ruby's method lookup structure to find and define the
# necessary methods for properties. This is accomplished through the method
# method_missing and define_method.
#
# This should be a consideration if there is a concern about the performance of
# the objects that are created, as there is much more overhead in the setting
# of these properties compared to using a Hash or a Struct.
```

TLDR; Use DeepStruct with care in controlled places (such as config is one example but not the only use case). I was even thinking of making this into its own gem for future re-use.

Just my 2c

**Comment by martinsvalin** [comment]
Ok, so maybe document the proper use-cases of the class at the top of the file. Also, there's already both a `deep_struct` and `deepstruct` gem. Let's not add to the confusion. ;)
